### DISCUSSION TITLE
A Simple Example System for Understanding OSCAL Components

### DISCUSSION THREAD

**Discussion by gregelin** [usnistgov/OSCAL#1177] [Q&A]
## About

**My hope for OSCAL:** 

IT Systems can reuse components to go from Vendor to Assessments reusing as much information as possible, especially control implementation statements.

**My fear for OSCAL:** 

Divergent definitions for components results limit us to today’s status quo reuse only of common controls unique to each organization that cannot be decomposed into specific technologies.

**For discussion:** 

What can we say/agree about the components and reuse of the respective control implementation statements of the below Simple Example Site?  (Questions follow example's description.)

## Simple Example Site IT System

The “Simple Example Site” (SES or “SimpleSite”) IT System consisting of a Django web site running on a single virtual host using the RHEL 8 at a federal agency which has defined it’s FISMA-related IT policies and provides a couple core common controls for authentication and logging.

```
Name:           Simple Example Site
Environment:    Agency’s Enterprise Data Center’s (EDC) PaaS Linux Server
Host OS:        RHEL 8
Database:       SQLite running on same server
Policies:       Agency’s policies for all systems
Authentication: Agency’s Enterprise Authentication Service (EAS)
Baseline:       NIST 800-53 Rev 5, Low Impact
```
##  Common Controls

```
Agency Policies
Enterprise Data Center’s PaaS Linux Server
Enterprise Logging Service
Enterprise Authentication Service
```

## Software

```
OS:  Linux RHEL 8
App: Django (Python-based application framework)
EDC: VMWare under the hood
EAS: OKTA OIDC under the hood
ELS: Splunk under the hood
```

## Questions

How many components are in the system?

- 4 (just common controls)
- 5 (common controls + system)
- 8 (Common controls + software)
- 9 (Common controls + software + policies)
- 10 (Common controls + software + policies + system)
- More?
- 11+ (Common controls + Software + policies + web site system + instance of linux server + other instances + workstation accessing system?)

**What are the types of components in the System?**

**To what degree can control implementation statements provided by the vendor be reused?**

- None - only common controls are reusable components and the implementation is too unique at each agency
- Little - As a draft content, but has to be re-written for the common controls
- Extensively - The common controls can/should combine content from the vendor supplied content

**What are the ATO/assessment boundaries relevant to the system?**

**What would be the OSCAL representing the system?**

**If the answers to any of the above are, “It depends,” then what is the criteria that determines different resulting answers (e.g., what decisions is the agency making that determines a different “correct” variation)?**

_Feel free to weigh in on any or all of these questions or share another observation._

**Comment by iMichaela** [comment]
@gregelin - You are listing the following "common controls" which are not controls ...
- Enterprise Data Center’s PaaS Linux Server
- Enterprise Logging Service
- Enterprise Authentication Service
They are services providing the implementation/satisfaction of more than 3 controls! The listed services could be standalone systems that have their own ATOs, systems that provide the implementation of many IA and AU controls respectively, as described in their functionality (e.g. what authentication methods are supported, enrollment process, etc. or audit record generation, protection, format) 
The PaaS Linux server can be a separate leveraged ATO or you can extend the authorization boundary to cover the PaaS as well. Your choice.
My 2 cents.

**Comment by david-waltermire** [comment]
A quick clarification first. Agency Policies might not be a system (with SSP) per se. These could be defined as a components of type=”policy” in a component definition. One per policy. You could also group them as a single component, but the more granular component per policy is better.

The following is a way to do the math for the “Simple Site System” SSP, which would have the following component considerations:

1.	Agency Policy Component(s): For all applicable policies there would be one component in the SSP per component from the component definitions. Note: If a given policy is implemented in multiple ways in the system, this would result in multiple SSP components that would use the same component from the component definitions as a base. I don’t think this is the case here.
2.	Enterprise Data Center’s PaaS Linux Server: This could be a component in the "Common Controls SSP".
3.	Enterprise Logging Service: This could be a component in the "Common Controls SSP".
4.	Enterprise Authentication Service: This could be a component in the "Common Controls SSP".

The following are my assumptions about the referenced OSCAL documentation.

- The "Common Controls" implementations will be in separate SSPs. These can be combined into a single SSP, or separated based on organizational preferences. I am choosing to separate them.
- In some cases the component type might be "software" or "service". To some degree it matters how the software is released and managed. If it is a standalone piece of software then the "software" type is appropriate. If it is only provided as a service to interact with, then the "service" type would be appropriate. I think in all cases below, "software" is the right choice.

# OSCAL Component Definition

Agency Policy Component Definition
- Components
   - policy-a (type=”policy”) – the policy to be implemented

# OSCAL System Security Plans

## Enterprise Data Center System

The "Enterprise Data Center System" will have at minimum the following components.

- edc-system (type=”this-system”) – Used for controls implemented at the system level not related to a specific component of the system.
- rhel-8-paas (type=”software”) – Used for controls implemented for the RHEL 8 PaaS.
- vmware (type=”software”) – Used for controls implemented for VMWare

## Enterprise Logging Service System

The "Enterprise Logging Service System" will have at minimum the following components.

- els-system (type=”this-system”) – Used for controls implemented at the system level not related to a specific component of the system.
- splunk (type=”software”) – Used for controls implemented for Splunk

If this is hosted in the "Enterprise Data Center System", then an OSCAL "leveraged-authorization" object can be used to point to the "Enterprise Data Center System" SSP and individual components can be defined to reference the "rhel-8-paas" or "vmware" software components provided by this common control provider. I omitted this for simplicity sake.

## Enterprise Authentication Service System

The "Enterprise Authentication Service System" will have at minimum the following components.

- eas-system (type=”this-system”) – Used for controls implemented at the system level not related to a specific component of the system.
- okta-oidc (type=”software”) – Used for controls implemented in the OKTA OIDC solution.

If this is hosted in the "Enterprise Data Center System", then an OSCAL "leveraged-authorization" object can be used to point to the "Enterprise Data Center System" SSP and individual components can be defined to reference the "rhel-8-paas" or "vmware" software components provided by this common control provider. I omitted this for simplicity sake.

## Simple Site System

The "Simple Site System" will define OSCAL "leveraged-authorization" objects for the following common control provider systems.

- Enterprise Data Center System
- Enterprise Logging Service System
- Enterprise Authentication Service System

The "Simple Site System" will have at minimum the following components.

1. sss-system (type=”this-system”) – Used for all controls implemented by this system not allocated to a specific component.
2. policy-a (type=”policy”) – Uses the information from the corresponding component definition
3. edc-system (type=”system”) - Leveraging the component from the Enterprise Data Center System
4. rhel-8-paas (type=”software”) – Leveraging the component from the Enterprise Data Center System
5. els-system (type=”system”) – Leveraging the component from the Enterprise Logging Service System
6. eas-system (type=”system”) – Leveraging the component from the Enterprise Authentication Service System
7. sqlite (type=”software”)
8. python (type=”software”)
9. django (type=”software”)

The "Simple Site System" could also reference the following components (in their respective systems) if any controls are exported/inherited for/from these components. I wasn’t sure if this was the case or not. These components could originally be defined in component definition(s) as well, which could be referenced at the point of initial use in the common control systems.

10. vmware
11. okta-oidc
12. splunk

# Answers to the Questions

> How many components are in the system?

So by my count you would have 9 (or maybe 12) components depending on the considerations above.

> What are the types of components in the System?

I indicated this above relative to each component.

> To what degree can control implementation statements provided by the vendor be reused?

There is not enough information in the scenario you provided to answer this. To some extent the answer depends on the quality of the vendor statements and their applicability to the risk appetite of the organization maintaining these information systems. OSCAL does not impose any limitations here. Vendor statements in component definitions can be reused to the extent that the organization maintaining these systems wants to. Vendor statements may be used wholesale, modified, or replaced on a case-by-case basis.

> What are the ATO/assessment boundaries relevant to the system?

This is not something that OSCAL is opinionated about. This is an organizational decision. Organizations should consider the appropriate guidelines. OSCAL is not and will not be prescriptive here. OSCAL simply provides a place to document the organizational decisions.

Where OSCAL can help with the ATO process is by defining where and how control implementation details are inherited (using the leveraged-authorization object), which can be used by an authorizing official (and their delegates) in deciding to issue an ATO. This view of inheritance helps them understand what controls and risks are addressed by the combined solution and what due diligence has been done by others in the systems their systems are inheriting from.

> What would be the OSCAL representing the system?

This can be built, at least partially. There is no control implementation narrative included here, so this would need to be stubbed out to make valid OSCAL SSPs. I'll let this discussion continue for a bit. Maybe someone will volunteer to build the examples.

**Comment by iMichaela** [comment]
@david-waltermire-nist - this is a great guidance for the problem stated. 
@gregelin - there is only one thing I would highlight related to what David wrote:

>The "Simple Site System" could also reference the following components (in their respective systems) if any controls are exported/inherited for/from these components. I wasn’t sure if this was the case or not. These components could originally be defined in component definition(s) as well, which could be referenced at the point of initial use in the common control systems.
>   10. vmware
>   11. okta-oidc
>   12. splunk

They can be treated as separate systems that are ATOed separately and which provide common controls implementations to other systems. You can think of OKTA a system on its own that has controls implemented (IA included) to secure access to the OKTA's data, the identities and the credentials managed. The IA controls provided to other systems are not the same instances of the controls implemented to secure OKTA (they are not inherited in the sense that if you get authenticated to access OKTA-system you also get authenticated to access other systems for which OKTA provides ID-as-a-Service. For example admins managing OKTA might need multi factors authentication, but OKTA could just provide password-based authentication to other systems.

There are controls like physical security controls that are inherited by all systems residing in that facility because once the facility is locked, the control is applicable through inheritance to the systems located there.

OSCAL facilitates the representation of the information but neither OSCAL not the RMF teams are imposing on organizations or system owners how to compose or decompose their systems and networks to perform a proper assessment or to ConMon the system. The only rule I recall from the RMF 800-37 rev2 is that everything that it is brought into the authorization boundary of a system must be assessed and authorization as part of that system. So if splunk becomes part of each system it is monitoring, it has to be assessed that it is securely managing data collected with every system it becomes part of, triggering redundant assessment work. If it is treated as a separate system that provides monitoring controls to other systems, only the implementation of the controls provided to other systems need to be assessed during system's assessment, and the ATO of the splunk system is used in the process of assessing the risk and authorizing the system using the controls provided by splunk.

**Comment by iMichaela** [comment]
Similar to the Splunk example used above would be Active Directory.
An organization can proceed as described below, with the Active Directory a component of the Common Controls SSP.

|Active Directory SSP              |   Common Controls SSP                 |   Other System using SSO  |
|------------------------------|------------------------------------|--------------------------------------|
| Describes how the system   Is secured. The system gets ATOed with a different cadence than the other systems using the controls provided. System owners and authorizing officials are in general different than for the systems using SSO  |  Describes among others,  the SSO controls implemented by the Active Dir system that becomes a component of the Comm Con SSP and uses ‘export’ and  ‘provided’ constructs for Controls since they are inherited. The grouping of all controls provided by the Active directory under a component will allow for traceability and assessment planning  |      Inherits controls from Common Controls SSP in addition to the locally implemented controls |
